tests/integration/components/key-press-listener-test.js

```diff
@@ -0,0 +1,48 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import {
+  initialize,
+  triggerKeyDown,
+  triggerKeyPress,
+  triggerKeyUp,
+} from 'ember-keyboard';
+
+module('Integration | Component | key-press-listener', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+    initialize();
+  });
+
+  test('it can handle keyDown', async function(assert) {
+    this.set('onKeyDown', function() {
+      assert.step('key-down');
+    });
+    await render(hbs`{{key-press-listener key="KeyA" onKeyDown=onKeyDown}}`);
+
+    triggerKeyDown('KeyA');
+    assert.verifySteps(['key-down'], 'key down should work');
+  });
+
+  test('it can handle keyUp', async function(assert) {
+    this.set('onKeyUp', function() {
+      assert.step('key-up');
+    });
+    await render(hbs`{{key-press-listener key="KeyA" onKeyUp=onKeyUp}}`);
+
+    triggerKeyUp('KeyA');
+    assert.verifySteps(['key-up'], 'key up should work');
+  });
+
+  test('it can handle keyPress', async function(assert) {
+    this.set('onKeyPress', function() {
+      assert.step('key-press');
+    });
+    await render(hbs`{{key-press-listener key="KeyA" onKeyPress=onKeyPress}}`);
+
+    triggerKeyPress('KeyA');
+    assert.verifySteps(['key-press'], 'key press should work');
+  });
+});
```

Comment:
These tests are good, but you're only testing the basic and happy path - what if e.g. `onKeyUp` is updated conditionally when a flag changes (e.g. `onKeyUp=(if someFlag (action "action1") (action "action2")`)? How about when multiple event handlers are passed into the same `key-press-listener` instance?

Reply:
👍 will add tests to cover the multiple listeners scenario. 

The conditional action scenario sounds interesting. How would that affect a component?

Reply:
At the moment, because the listeners get wired up in `didInsertElement`, if the passed-in action changes as I described above, what happens? Does the listener automagically trigger the new action, or will it continue to trigger the old one?

There's also a more subtle possibility of this case arising too, where someone is using currying on actions, or a helper that returns an action (e.g. `ember-concurrency`'s `perform` helper); in that case, any time one of the parameters passed into the currying/helper changes, a new action will be generated, and if the listener doesn't support hot-swapping that action then there could be subtle bugs happening as a result (e.g. an action continues to get called with a previous value instead of an updated one).

These possibilities aren't really much of a concern in our use case here, but this component seems pretty generic so I was thinking about possible re-use scenarios 🤷‍♂️